themes/qgrid/source/js/menu.js

```diff
@@ -15,26 +18,231 @@ function toggleVisibility(e) {
 	}
 }
 
-function filter(e) {
+function onSearchChange(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
-	const filter = input.value.toLowerCase();
+	const search = input.value.toLowerCase();
+	setSearch(search);
+}
+
+function searchOnLoad() {
+	const active = document.querySelector('.active-topic');
+	const search = getSearch();
+
+	if (active && search) {
+		document.getElementById('search').value = search;
+		currentPageSearch(search);
+		menuItemsSearch(search);
+	}
+}
+
+function searchOnHashChange() {
+	const search = getSearch();
+	document.getElementById('search').value = search;
+
+	if (!search) {
+		document.getElementById('search').focus();
+	}
+	currentPageSearch(search);
+	menuItemsSearch(search);
+}
+
+function currentPageSearch(search) {
+	let scrolled = false;
+
+	document.querySelectorAll('.page-wrapper *').forEach(node => {
+		if (SEARCHEABLE_ELEMENTS.includes(node.nodeName)) {
+			const a = node.querySelector('a:last-child');
+			const pageItem = (a && a.textContent) ? a : node;
+			const { textContent } = pageItem;
+			const words = splitToWords(search);
+			const firstMatch = (testSearch(textContent, search)) ? words[indexOfSearch(textContent, words)] : '';
+			const searchTarget = (textContent.toLowerCase().includes(search)) ? search : firstMatch;
+
+			if (highlightText(pageItem, searchTarget) && !scrolled) {
```

Comment:
it's not always necessary to add newline, just follow that it should be look good like in book